docker-compose.yml

```diff
@@ -15,732 +15,779 @@
 
-  cpp-minimum-boost-version:
-    # Ubuntu Xenial 16.04 has Boost 1.58, our minimum version
-    #
+  ubuntu-cpp-cmake32:
     # Usage:
-    #   docker-compose build cpp-minimum-boost-version
-    #   docker-compose run cpp-minimum-boost-version
-    image: arrowdev/arrow-cpp-ubuntu-xenial:latest
-    shm_size: 2G
+    #   docker-compose build ubuntu-cpp
+    #   docker-compose build ubuntu-cpp-cmake32
+    #   docker-compose run ubuntu-cpp-cmake32
+    image: ${ORG}/${ARCH}-ubuntu-${UBUNTU}-cpp-cmake-3.2:latest
     build:
       context: .
-      dockerfile: cpp/Dockerfile.ubuntu-xenial
+      cache_from:
+        - ${ORG}/${ARCH}-ubuntu-${UBUNTU}-cpp-cmake-3.2:latest
+      dockerfile: ci/docker/linux-apt-cmake.dockerfile
+      args:
+        base: ${ORG}/${ARCH}-ubuntu-${UBUNTU}-cpp:latest
+        cmake: 3.2.3
     environment:
-      PARQUET_TEST_DATA: /arrow/cpp/submodules/parquet-testing/data
-      ARROW_BOOST_SOURCE: "SYSTEM"
+      # Vendor boost to avoid dealing with stale FindBoost.
+      ARROW_BOOST_SOURCE: "BUNDLED"
+      double-conversion_SOURCE: "BUNDLED"
+      ARROW_ORC: "OFF"
+      ARROW_GANDIVA: "OFF"
+      # uriparser requires cmake 3.3
       ARROW_FLIGHT: "OFF"
-    volumes: *ubuntu-xenial-volumes
+    volumes: *ubuntu-volumes
+    command: *cpp-command
 
-  cpp-ubuntu-bionic:
-    # Usage:
-    #   docker-compose build cpp-ubuntu-bionic
-    #   docker-compose run cpp-ubuntu-bionic
-    image: arrowdev/arrow-cpp-ubuntu-bionic:latest
-    shm_size: 2G
+  conda-python-pandas:
+    # document to use --no-cache in case of PANDAS=master
+    image: ${ORG}/${ARCH}-conda-latest-python-${PYTHON}-pandas-${PANDAS}:latest
     build:
       context: .
-      dockerfile: cpp/Dockerfile.ubuntu-bionic
+      dockerfile: ci/docker/conda-latest-python-pandas.dockerfile
+      cache_from:
+        - ${ORG}/${ARCH}-conda-latest-python-${PYTHON}-pandas-${PANDAS}:latest
+      args:
+        org: ${ORG}
+        arch: ${ARCH}
+        conda: latest
+        python: ${PYTHON}
+        pandas: ${PANDAS}
+    shm_size: *shm-size
+    volumes: *conda-volumes
+    entrypoint: *conda-entrypoint
+    command: *python-conda-command
+
+  conda-python-dask:
+    # document to use --no-cache in case of DASK=master
+    image: ${ORG}/${ARCH}-conda-latest-python-${PYTHON}-dask-${DASK}:latest
+    build:
+      context: .
+      dockerfile: ci/docker/conda-latest-python-dask.dockerfile
+      cache_from:
+        - ${ORG}/${ARCH}-conda-latest-python-${PYTHON}-dask-${DASK}:latest
+      args:
+        org: ${ORG}
+        arch: ${ARCH}
+        conda: latest
+        python: ${PYTHON}
+        dask: ${DASK}
+    shm_size: *shm-size
+    volumes: *conda-volumes
+    entrypoint: *conda-entrypoint
+    command:
+      ["/arrow/ci/scripts/cpp_conda_build.sh /arrow /build/cpp &&
+        /arrow/ci/scripts/python_build.sh /arrow &&
+        /arrow/ci/scripts/integration_dask.sh"]
+
+  conda-python-turbodbc:
```

Comment:
@xhochy does this need to live in arrow, can we move it in turbodbc? The goal would be to lower the maintenance burden.